advance-exercises/src/components/04-OptimizeRender/OptimizeRender.js

```diff
@@ -23,11 +23,18 @@ class ListView extends React.Component {
     rowHeight: PropTypes.number.isRequired,
     renderRowAtIndex: PropTypes.func.isRequired,
   };
-
+  componentDidMount() {
+    document.getElementById('content').addEventListener('scroll', this.handleScroll);
+    this.setState({ ...this.state, numRows: Math.floor(document.getElementById('content').clientHeight / this.props.rowHeight) })
```

Comment:
I feel that there are too many operations happening on the same line. Do consider separating them for more readability.